distribution/sdk/sample-mpegts-streamgenerator/src/main/java/org/codice/alliance/distribution/sdk/video/stream/mpegts/MpegTsUdpClient.java

```diff
@@ -0,0 +1,305 @@
+            } catch (NumberFormatException e) {
+                LOGGER.warn("Unable to parse specified port : {}.  Using Default : {}",
+                        arguments[2],
+                        DEFAULT_PORT);
+                port = DEFAULT_PORT;
+            }
+        }
+
+        LOGGER.info("Video file path : {}", videoFilePath);
+
+        LOGGER.info("Streaming address : {}:{}", ip, port);
+
+        final AtomicLong count = new AtomicLong(0);
+
+        EventLoopGroup eventLoopGroup = new NioEventLoopGroup();
+        try {
+
+            Bootstrap bootstrap = new Bootstrap();
+
+            bootstrap.group(eventLoopGroup)
+                    .channel(NioDatagramChannel.class)
+                    .option(ChannelOption.SO_BROADCAST, true)
+                    .handler(new SimpleChannelInboundHandler<DatagramPacket>() {
+                        @Override
+                        protected void channelRead0(ChannelHandlerContext channelHandlerContext,
+                                DatagramPacket datagramPacket) throws Exception {
+                            LOGGER.info("Reading datagram from channel");
+                        }
+
+                        @Override
+                        public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause)
+                                throws Exception {
+                            logErrorMessage(String.format(
+                                    "Exception occured handling datagram packet.  %s",
+                                    cause), false);
+                            ctx.close();
+                        }
+                    });
+
+            Channel ch = bootstrap.bind(0)
+                    .sync()
+                    .channel();
+
+            File videoFile = new File(videoFilePath);
+
+            long bytesSent = 0;
+
+            long tsPacketCount = videoFile.length() / PACKET_SIZE;
+
+            Duration videoDuration = getVideoDuration(videoFilePath);
+            if (videoDuration == null) {
+                return;
+            }
+
+            long tsDurationMillis = videoDuration.toMillis();
+
+            LOGGER.info("Video Duration : {}", tsDurationMillis);
+
+            double delayPerPacket = (double) tsDurationMillis / (double) tsPacketCount;
+
+            long startTime = System.currentTimeMillis();
+
+            int packetsSent = 0;
+
+            try (final FileInputStream fis = new FileInputStream(videoFile)) {
+
+                byte[] buffer = new byte[PACKET_SIZE];
+                int c;
+                while ((c = fis.read(buffer)) != -1) {
+                    bytesSent += c;
+
+                    ChannelFuture cf = ch.writeAndFlush(new DatagramPacket(Unpooled.copiedBuffer(
+                            buffer), new InetSocketAddress(ip, port)));
+
+                    cf.await();
+
+                    packetsSent++;
+
+                    if (packetsSent % 100 == 0) {
```

Comment:
Any particular reason you're modding on 100 here, or just an arbitrary number of packets between delays?